### PR TITLE
Spec for opcode `ADDRESS`

### PR DESCRIPTION
**Pull request by silathdiir**
Circuit: https://github.com/scroll-tech/zkevm-circuits/pull/181

Mainly reference `caller.py`.

### CODE REVIEW

File: specs/opcode/30ADDRESS.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+3. Do busmapping lookup for stack write operation
+
+## Constraints
+
+1. opId = 0x30
+2. State transition:
+   - gc + 2 (1 stack write, 1 call context read)
+   - stack_pointer - 1
+   - pc + 1
+   - gas + 2
```
Comment (DreamWuGit) [inline specs/opcode/30ADDRESS.md:25]: gas - 2 ?

**Review by DreamWuGit — APPROVED** [review]
LGTM after comment fixed

File: specs/opcode/30ADDRESS.md (hunk 1 above)
Comment (silathdiir) [reply specs/opcode/30ADDRESS.md]: Fixed to `gas - 2`. Thanks.

**Comment by silathdiir** [comment]
Thanks @DreamWuGit 's code review. Moved to upstream: https://github.com/privacy-scaling-explorations/zkevm-specs/pull/247

**Pull request closed without merging**